samples/stock/demo/pointandfigure/demo.js

```diff
@@ -240,16 +237,22 @@
             takeOrdinalPosition: true,
             finalData: [],
             getColumnMetrics:
-            Highcharts.seriesTypes.column.prototype.getColumnMetrics,
+                Highcharts.seriesTypes.column.prototype.getColumnMetrics,
             markerAttribs,
             generatePnfData,
             translate() {
                 const metrics = this.getColumnMetrics(),
                     calculatedBoxSize = this.calculatedBoxSize;
-                this.markerWidth =
-                metrics.width + metrics.paddedWidth + metrics.offset;
+
                 this.markerHeight =
-                this.yAxis.toPixels(0) - this.yAxis.toPixels(calculatedBoxSize);
+                    this.yAxis.toPixels(0) -
+                    this.yAxis.toPixels(calculatedBoxSize);
+
+                this.markerWidth = this.options.maxPointWidth || Math.min(
+                    this.markerHeight * 2,
+                    metrics.width + metrics.paddedWidth + metrics.offset
```

Comment:
It looks incorrect in mobile view, markers overlap. Could you reduce the calculation to `metrics.width`?